### PR TITLE
do not blindly override permission set of ldap users

### LINKED ISSUE #2516: Missing setup streams and user dashboards after entering through the active directory
### Problem description

After obnovlenyai from 1.7 to 2.0 version, settings are lost after entering through Active Directory. So it should be, or am I doing something wrong? Previously, it was possible to adjust the stream and dashboards for each individual.

### Steps to reproduce the problem

1. The user comes to the AD account
2. I add it streams and dashboards
3. After you log out and log in, setting the empty again

### Environment

* Graylog Version: 2.0
* Elasticsearch Version: 2.3.4
* MongoDB Version: 3.2.8
* Operating System: ubuntu 14.04
* Browser version: Chrome

### PR DESCRIPTION
**Pull request by kroepke**
when fixing #2045 we introduced a bug clearing the permission set for synced ldap accounts

this only shows up when not using roles but the permission sets directly

fixes #2516

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/users/UserImpl.java [hunk 1]
```diff
@@ -172,8 +173,9 @@ public void setEmail(final String email) {
 
     @Override
     public void setPermissions(final List<String> permissions) {
+        final List<String> perms = Lists.newArrayList(permissions);
         // Do not store the dynamic user self edit permissions
-        permissions.removeAll(this.permissions.userSelfEditPermissions(getName()));
+        perms.removeAll(this.permissions.userSelfEditPermissions(getName()));
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/users/UserImpl.java]: I can has unit test for this?

Comment (kroepke) [inline graylog2-server/src/main/java/org/graylog2/users/UserImpl.java]: touché

File: graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java [hunk 2]
```diff
@@ -250,7 +249,8 @@ private void updateFromLdap(User user, LdapEntry userEntry, LdapSettings ldapSet
             translatedRoleIds.addAll(user.getRoleIds());
         }
         user.setRoleIds(translatedRoleIds);
-        user.setPermissions(Collections.emptyList());
+        // preserve the raw permissions (the ones without the synthetic self-edit permissions or the "*" admin one)
+        user.setPermissions(user.getPermissions());
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java]: I can has unit test for this?

**Comment by joschi** [comment]
LGTM. 👍

**Comment by angelsedoy** [comment]
I'm sorry, but this code does not work. Now permissions are not overwritten, but are overwritten role, in particular the role of flies admin.

**Comment by joschi** [comment]
@angelsedoy Thanks for your feedback. Please create a new issue for this at https://github.com/Graylog2/graylog2-server/issues/new

**Pull request merged**